pkg/ruler/api.go

```diff
@@ -0,0 +1,111 @@
+package ruler
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/gorilla/mux"
+
+	"github.com/weaveworks/common/user"
+	"github.com/weaveworks/cortex/pkg/configs"
+	"github.com/weaveworks/cortex/pkg/configs/db"
+	"github.com/weaveworks/cortex/pkg/util"
+)
+
+// API implements the configs api.
+type API struct {
+	db db.RulesDB
+	http.Handler
+}
+
+// NewAPIFromConfig makes a new API from our database config.
+func NewAPIFromConfig(cfg db.Config) (*API, error) {
+	db, err := db.NewRulesDB(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return NewAPI(db), nil
+}
+
+// NewAPI creates a new API.
+func NewAPI(db db.RulesDB) *API {
+	a := &API{db: db}
+	r := mux.NewRouter()
+	a.RegisterRoutes(r)
+	a.Handler = r
+	return a
+}
+
+// RegisterRoutes registers the configs API HTTP routes with the provided Router.
+func (a *API) RegisterRoutes(r *mux.Router) {
+	for _, route := range []struct {
+		name, method, path string
+		handler            http.HandlerFunc
+	}{
+		{"get_rules", "GET", "/api/prom/rules", a.getConfig},
+		{"set_rules", "POST", "/api/prom/rules", a.setConfig},
+	} {
+		r.Handle(route.path, route.handler).Methods(route.method).Name(route.name)
+	}
+}
+
+// getConfig returns the request configuration.
+func (a *API) getConfig(w http.ResponseWriter, r *http.Request) {
+	userID, _, err := user.ExtractOrgIDFromHTTPRequest(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+	logger := util.WithContext(r.Context(), util.Logger)
+
+	cfg, err := a.db.GetRulesConfig(userID)
+	if err == sql.ErrNoRows {
+		http.Error(w, "No configuration", http.StatusNotFound)
+		return
+	} else if err != nil {
+		// XXX: Untested
```

Comment:
What is the expectation with this comment?

Reply:
It's cargo-culted from the config API. The vague expectation is that we'd test the error path one of these days. I've removed it & its like.